File: ASSIGNMENT3/EmployeeManagementSystem/src/exceptions/InvalidDateException.java
package exceptions;

public class InvalidDateException extends Exception {
    public InvalidDateException(String message) {
        super(message);
    }

    public static void DateValidation(String date) {
        try {
            if (date == null || date.trim().isEmpty()) {
                throw new InvalidDateException("Date cannot be null or empty.");
            }
            if (!date.matches("\\d{2}-\\d{2}-\\d{4}")) {
                throw new InvalidDateException("Date must be in the format DD-MM-YYYY.");
            }
            String[] parts = date.split("-");
            // int year = Integer.parseInt(parts[0]);
            int month = Integer.parseInt(parts[1]);
            int day = Integer.parseInt(parts[2]);
            
            if (month < 1 || month > 12) {
                throw new InvalidDateException("Bad month: " + month);
            }
            if (day < 1 || day > 31) {
                throw new InvalidDateException("Bad day: " + day);
            }
        } catch (InvalidDateException e) {
            System.err.println("Error: " + e.getMessage());
        }
    }
    
}

File: Advanced_Prog_Assignment3-csharp/EmployeeManagementSystem/Exceptions/InvalidDateException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Text.RegularExpressions;

namespace EmployeeManagementSystemCSharp.Exceptions
{
    public class InvalidDateException : Exception
    {
        public InvalidDateException(string message)
            : base(message)
        {
        }

        public static void DateValidation(string date)
        {
            try
            {
                if (date == null || date.Trim().Length == 0)
                {
                    throw new InvalidDateException("Date cannot be null or empty.");
                }

                if (!Regex.IsMatch(date, @"^\d{2}-\d{2}-\d{4}$"))
                {
                    throw new InvalidDateException("Date must be in the format DD-MM-YYYY.");
                }

                string[] parts = date.Split('-');

                int month = int.Parse(parts[1]);
                int day = int.Parse(parts[2]);

                if (month < 1 || month > 12)
                {
                    throw new InvalidDateException("Bad month: " + month);
                }

                if (day < 1 || day > 31)
                {
                    throw new InvalidDateException("Bad day: " + day);
                }
            }
            catch (InvalidDateException e)
            {
                Console.Error.WriteLine("Error: " + e.Message);
            }
        }
    }
}
